### DISCUSSION TITLE
Running a Python script through 'bash -c' in cmd get UnicodeEncodeError

### DISCUSSION THREAD

**Discussion by Leoongithub** [microsoft/WSL#6256] [General]
# Environment

```none
Windows build number: [Windows 10.0.19042.630]
Your Distribution version: [Ubuntu 18.04]
Whether the issue is on WSL 2 and/or WSL 1: [WSL 1]
```

# Steps to reproduce
When I run a Python script through `bash -c` in `cmd`, Python throw UnicodeEncodeError. But in wsl it runs correctly.

![image](https://user-images.githubusercontent.com/10937162/99543080-b7a58c80-29ed-11eb-8b60-417b1ee5ee3a.png)

#  Expected behavior

Print Unicode character

# Actual behavior

Python throw UnicodeEncodeError

**Comment by therealkenc** [comment]
This error is from userspace (ie Python). The `bash.exe` command has been deprecated; use `wsl.exe`. It is probable that the problem is related to your Python configuration living in `.bashrc` (or similar), which is not being run in the above command. Try something like:
```
C:\> wsl.exe /bin/bash -ic /home/you/test.py
```

**Reply by Leoongithub** [reply]
I tried your command and got the same result. Still thanks for your help.

**Comment by therealkenc** [comment]
Thanks for the follow up. There isn't enough here for me to reproduce the problem locally. I am not familiar enough with Python to tell you with specificity what causes Python to output `codec can't encode character`, but it appears to be something related to language settings.

**Comment by Leoongithub** [comment]
Still have no idea to solve the problem. But finally I found aother way to print unicode string:

```
import sys
sys.stdout.buffer.write('啊'.encode('utf-8'))
```

Maybe it is not elegant but it works.
